kibble/models/item_index.go

```diff
@@ -83,6 +83,25 @@ func (itemIndex ItemIndex) Set(slug string, item GenericItem) {
 	}
 }
 
+// Set - an item
+func (itemIndex ItemIndex) SetWithStatus(slug string, statusID int, item GenericItem) {
+
+	slugType := getSlugType(slug)
+
+	index, ok := itemIndex[slugType]
+	if !ok {
+		itemIndex[slugType] = make(map[string]GenericItem)
+		index = itemIndex[slugType]
+	}
+
+	// unresolved can be overwritten, empty and item can not
+	foundItem, ok := index[slug]
+	item.StatusID = statusID
+	if (ok && (foundItem == Unresolved || foundItem == Empty)) || !ok {
```

Comment:
Is this the same?

```suggestion
	if !ok || foundItem == Unresolved || foundItem == Empty {
```

Reply:
Also from the Set function, but yeah that way does seem a lot more readable, will update